### PR TITLE
home-assistant: support onewire component

### PR DESCRIPTION
**Pull request by dotlambda**
###### Motivation for this change
support 80% of all Home Assistant components :tada: 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 159906` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>python310Packages.pi1wire</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>home-assistant-component-tests.onewire</li>
    <li>python310Packages.pyownet</li>
    <li>python39Packages.pi1wire</li>
    <li>python39Packages.pyownet</li>
  </ul>
</details>

File: pkgs/development/python-modules/pi1wire/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    pytestCheckHook
+  ];
+
+  pythonImportsCheck = [ "pi1wire" ];
+
+  meta = with lib; {
+    description = "1Wire Sensor Library for Raspberry PI";
+    homepage = "https://github.com/ushiboy/pi1wire";
+    license = licenses.mit;
+    maintainers = with maintainers; [ dotlambda ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/pi1wire/default.nix:30]:
```suggestion
    maintainers = with maintainers; [ dotlambda ];
    broken = pythonAtLeast "3.10";
```

**Comment by dotlambda** [comment]
> 1 package failed to build

can't reproduce

**Comment by mweinelt** [comment]
```
=================================== FAILURES ===================================
____________________________ test_find_all_sensors _____________________________

temp_dir_path = '/build/570e3b80-832c-411a-bca9-3292a0bb366e'

    def test_find_all_sensors(temp_dir_path):
        os.mkdir(os.path.join(temp_dir_path, '28-000000654321'))
        os.mkdir(os.path.join(temp_dir_path, '28-000000987654'))

        p = Pi1Wire(base_path=temp_dir_path)
        r = p.find_all_sensors()
        assert len(r) == 2
>       assert r[0].mac_address == '28000000654321'
E       AssertionError: assert '28000000987654' == '28000000654321'
E         - 28000000654321
E         ?            ---
E         + 28000000987654
E         ?         +++

tests/test_finder.py:14: AssertionError
=========================== short test summary info ============================
FAILED tests/test_finder.py::test_find_all_sensors - AssertionError: assert '...
========================= 1 failed, 6 passed in 0.11s ==========================
builder for '/nix/store/shppw8nkwp2d8g7lakfljbxlr8scs31x-python3.10-pi1wire-0.2.0.drv' failed with exit code 1
````

**Comment by dotlambda** [comment]
@mweinelt I tried rebuilding it many times. Must be flaky

**Comment by mweinelt** [comment]
Are you trying the python310 version? Because I can't get it to pass the tests, tried on multiple machines.

@ofBorg build python310Packages.py1wire

**Comment by dotlambda** [comment]
Result of `nixpkgs-review pr 159906` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>home-assistant-component-tests.onewire</li>
    <li>python310Packages.pi1wire</li>
    <li>python310Packages.pyownet</li>
    <li>python39Packages.pi1wire</li>
    <li>python39Packages.pyownet</li>
  </ul>
</details>

**Comment by dotlambda** [comment]
Ofborg confirms the failure. Very mysterious.

**Comment by mweinelt** [comment]
Can't reproduce on aarch64-linux either. :laughing: 

Let's disable that test then.

**Pull request merged**